books/kestrel/c/syntax/types.lisp

```diff
@@ -39,111 +41,111 @@
-    (xdoc::li
-     "The three complex types [C17:6.2.5/11].
-      These are a conditional feature,
-      but they must be included in this fixtype
-      because this fixtype consists of all the possible types.")
-    (xdoc::li
-     "The @('_Bool') type [C17:6.2.5/2].")
-    (xdoc::li
-     "A family of structure types [C17:6.2.5/20].
-      Structure types are characterized by an optional tag.
-      This is an approximation,
-      because there may be different structure types of a given tag,
-      or different tagless structure types.")
-    (xdoc::li
-     "A collective type for all union types [C17:6.2.5/20].
-      This is an approximation,
-      because there are different union types.")
-    (xdoc::li
-     "A collective type for all enumeration types [C17:6.2.5/20].
-      This is an approximation,
-      because there are different enumeration types.")
-    (xdoc::li
-     "A collective type for all array types [C17:6.2.5/20].
-      This is an approximation,
-      because there are different array types.")
-    (xdoc::li
-     "A collective type for all pointer types [C17:6.2.5/20].
-      This is an approximation,
-      because there are different pointer types.")
-    (xdoc::li
-     "A collective type for all function types [C17:6.2.5/20].
-      This is an approximation,
-      because there are different function types.")
-    (xdoc::li
-     "An ``unknown'' type that we need due to our current approximation.
-      Our validator must not reject valid code.
-      But due to our approximate treatment of types,
-      we cannot always calculate a type,
-      e.g. for a member expression of the form @('s.m')
-      where @('s') is an expression with structure type.
-      Since our approximate type for all structure types
-      has no information about the members,
-      we cannot calculate any actual type for @('s.m');
-      but if the expression is used elsewhere,
-      we need to accept it, because it could have the right type.
-      We use this unknown type for this purpose:
-      the expression @('s.m') has unknown type,
-      and unknown types are always acceptable."))
-   (xdoc::p
-    "Besides the approximations noted above,
-     currently we do not capture atomic types [C17:6.2.5/20],
-     which we approximate as the underlying (argument) type.
-     We also do not capture @('typedef') names,
-     which we approximate as unknown types.
-     Furthermore, we do not capture qualified types [C17:6.2.5/26]."))
-  (:void ())
-  (:char ())
-  (:schar ())
-  (:uchar ())
-  (:sshort ())
-  (:ushort ())
-  (:sint ())
-  (:uint ())
-  (:slong ())
-  (:ulong ())
-  (:sllong ())
-  (:ullong ())
-  (:float ())
-  (:double ())
-  (:ldouble ())
-  (:floatc ())
-  (:doublec ())
-  (:ldoublec ())
-  (:bool ())
-  (:struct ((tag? ident-optionp)))
-  (:union ())
-  (:enum ())
-  (:array ())
-  (:pointer ())
-  (:function ())
-  (:unknown ())
-  :pred typep)
+(encapsulate ()
+  (set-induction-depth-limit 1)
```

Comment:
I would use a `:prepwork ((set-induction-depth-limit 1))` inside, at the end of, the `fty::deftagsum`, which seems more readable.